library/src/G4CMPVDriftProcess.cc

```diff
@@ -90,17 +90,18 @@ G4CMPVDriftProcess::PostStepGetPhysicalInteractionLength(
 // Fill ParticleChange energy and mass for electron charge carrier momentum
 
 void 
-G4CMPVDriftProcess::FillParticleChange(G4int ivalley, const G4ThreeVector& p) {
+G4CMPVDriftProcess::FillParticleChange(const G4ThreeVector& p) {
   // Compute kinetic energy from momentum for electrons or holes
   G4double energy = 0.;
+  G4int valley = FindNearestValley(GetLocalDirection(p));
   if (IsElectron()){
-    energy = theLattice->MapPtoEkin(ivalley, GetLocalDirection(p));
+    energy = theLattice->MapPtoEkin(valley, GetLocalDirection(p));
```

Comment:
With this change, we are going to need to re-run the samples and check energy conservation again.